Add Semgrep and Security Workflow

lindybritstravelperk opened the pull request:
- Adding Semgrep as part of the security checks for vulnerabilities. As part of any commits pushed, or any PRs made, Semgrep needs to run successfully in the GitHub workflow.

A new workflow (see [job](https://github.com/lindybritstravelperk/jets-seatmap-react-lib-pub/actions/runs/12909193742/job/35996325493)) that failed before `DOMPurify.sanitize` because of the 3 vulnerabilities detected by Semgrep:

![Screenshot 2025-01-22 at 14 40 48](https://github.com/user-attachments/assets/2629e17f-7e34-4ed4-8ad8-1b2dd454c6da)

After the fix to these 3 files, Semgrep does not report any more findings (see [job](https://github.com/lindybritstravelperk/jets-seatmap-react-lib-pub/actions/runs/12910500295/job/36000662662)):

![Screenshot 2025-01-22 at 14 42 54](https://github.com/user-attachments/assets/3f3542aa-4851-460d-949a-e49c3730ca33)

TravelPerk [JIRA](https://travelperk.atlassian.net/browse/APP-95509).

lindybritstravelperk committed:
Add Semgrep and Security Workflow

 - Adding Semgrep as part of the security checks for vulnerabilities. As part of any commits pushed, or any PRs made, Semgrep needs to run successfully in the GitHub workflow.

clicktravel-lorenamorales reviewed (approved):
Changes look good to me! Did you test the library after the changes? I guess that it works as expected right?

t-gallagher reviewed (approved):
LGTM

lindybritstravelperk commented:
> Changes look good to me! Did you test the library after the changes? I guess that it works as expected right?

Did a quick visual check in Storybook and all seemed normal from my side! The Quicket team might want to sense check it too, however.

lindybritstravelperk committed:
Replaced DOMPurify with Isomorphic DOMPurify

 - To make sure we don't break anything, we use the isomorphic-dompurify package instead, such that it can be used seamlessly on server and client in the same way (taking into account SSR).

nexus commented:
I'm not able to build the lib:
![image](https://github.com/user-attachments/assets/455c4d93-c641-4bf0-b522-d0aed4503ea7)
The new `isomorphic-dompurify` dependency causes an error when building lib, command `npm run build-lib`. 
It looks like you need to resolve the `rollup json` plugin dependency and configure `rollup` accordingly.

nexus reviewed (changes requested):
need to configure rollup with @rollup/plugin-json

nexus commented:
Also, I propose to add step `npm run build-lib` to the tests

lindybritstravelperk committed:
Review changes

nexus commented:
the size of the release version of the lib before this PR = 328K:
![image](https://github.com/user-attachments/assets/6af93ceb-35f1-4672-9e19-38c1b08efff5)
size after PR =  3.4M
![image](https://github.com/user-attachments/assets/096f13aa-6d03-4860-9816-69800472fb51)
I guess you need to find a better way to implement this `sanitize` feature

lindybritstravelperk commented:
> the size of the release version of the lib before this PR = 328K: ![image](https://private-user-images.githubusercontent.com/1560950/406012666-6af93ceb-35f1-4672-9e19-38c1b08efff5.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3Mzc2Mzc3MjAsIm5iZiI6MTczNzYzNzQyMCwicGF0aCI6Ii8xNTYwOTUwLzQwNjAxMjY2Ni02YWY5M2NlYi0zNWYxLTQ2NzItOWUxOS0zOGMxYjA4ZWZmZjUucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI1MDEyMyUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNTAxMjNUMTMwMzQwWiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9MTI0YzBkNTBiMDRhNTQwYjQwZTYzY2JhYjQwZjE0MjhiNmJiNzdlMTc2OTkyMDIwODE3N2MyMjdlYTg1OGE0ZCZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ.cs-KGVhNA3gMEmupinJLwDohIkpgxQQ8knNkkeD-Oks) size after PR = 3.4M ![image](https://private-user-images.githubusercontent.com/1560950/406013361-096f13aa-6d03-4860-9816-69800472fb51.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3Mzc2Mzc3MjAsIm5iZiI6MTczNzYzNzQyMCwicGF0aCI6Ii8xNTYwOTUwLzQwNjAxMzM2MS0wOTZmMTNhYS02ZDAzLTQ4NjAtOTgxNi02OTgwMDQ3MmZiNTEucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI1MDEyMyUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNTAxMjNUMTMwMzQwWiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9ZDhkMGZjMWEwMmJkZmRhM2E3OTRjZWJlOThjZDlmNzAyYTJlNzU4NDVlOWE3ZmFiMDgxYTZiNDI3MmU5MTUxYiZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ.CGAtHQj_IXI6L8EnsRTD9NPuJ8X2ZKJW7sPu7r1sOhU) I guess you need to find a better way to implement this `sanitize` feature

We'll take a look, thanks.

lindybritstravelperk commented:
> the size of the release version of the lib before this PR = 328K: ![image](https://private-user-images.githubusercontent.com/1560950/406012666-6af93ceb-35f1-4672-9e19-38c1b08efff5.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3Mzc2NDc2ODMsIm5iZiI6MTczNzY0NzM4MywicGF0aCI6Ii8xNTYwOTUwLzQwNjAxMjY2Ni02YWY5M2NlYi0zNWYxLTQ2NzItOWUxOS0zOGMxYjA4ZWZmZjUucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI1MDEyMyUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNTAxMjNUMTU0OTQzWiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9OWEyM2Q3MGQxYzQ2ZWQxZmJlYjQ5ODRjMDVjNjJhM2RhMzk4ZWZmYzA3MjgzY2QxMzEwMDI4ZWU2ODE5ZDJkZiZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ.4nq9nZZE0YlEOJLhipQWpjGhLlybSNLheEZqPoyofnc) size after PR = 3.4M ![image](https://private-user-images.githubusercontent.com/1560950/406013361-096f13aa-6d03-4860-9816-69800472fb51.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3Mzc2NDc2ODMsIm5iZiI6MTczNzY0NzM4MywicGF0aCI6Ii8xNTYwOTUwLzQwNjAxMzM2MS0wOTZmMTNhYS02ZDAzLTQ4NjAtOTgxNi02OTgwMDQ3MmZiNTEucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI1MDEyMyUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNTAxMjNUMTU0OTQzWiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9YjQ0OWY0MzI5M2U2NWZiMjk0ZWVmMzdhYjA1ODI3Y2JiMTAwZjgwNzY0OWQyNWZlODdhZjlkZjhlZGQ2OWQ5NyZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ.z-v4nFM4t45lG2gkXXV5dh663aiaR3N8HkRVJlYoCmM) I guess you need to find a better way to implement this `sanitize` feature

Changes have been made to `dompurify` to decrease the size. Also, the step `Test build-lib` was added into the test workflow.

<img width="469" alt="Screenshot 2025-01-23 at 15 29 36" src="https://github.com/user-attachments/assets/61d87ad0-4aaf-4f7b-b7d2-3a6dc6b487c5" />
